base/test/affinetransform_tests.cpp

```diff
@@ -0,0 +1,353 @@
+#include <boost/test/unit_test.hpp>
+#include "AffineTransform.h"
+#include "FileReaderModule.h"
+#include "RawImageMetadata.h"
+#include "FrameMetadata.h"
+#include "FrameMetadataFactory.h"
+#include "Frame.h"
+#include "Logger.h"
+#include "test_utils.h"
+#include "CudaMemCopy.h"
+#include "ExternalSinkModule.h"
+#include "AIPExceptions.h"
+#include "stdafx.h"
+#include "PipeLine.h"
+
+
+BOOST_AUTO_TEST_SUITE(affinetransform_tests)
+
+struct AffineTestsStruct
+{
+	std::string inpFilePath;
+	int width;
+	int height;
+	ImageMetadata::ImageType imageType;
+	int bit_depth;
+	int angle;
+	int x;
+	int y;
+	float scale;
+	framemetadata_sp metadata;
+
+	AffineTestsStruct(const std::string& inpFilePath, int width, int height, ImageMetadata::ImageType imageType, int bit_depth, int angle, int x, int y, float scale)
+	{
+		if (imageType == ImageMetadata::ImageType::YUV420 || imageType == ImageMetadata::ImageType::YUV444) {
+			metadata = framemetadata_sp(new RawImagePlanarMetadata(width, height, imageType, size_t(0), CV_8U));
+		}
+		else {
+			metadata = framemetadata_sp(new RawImageMetadata(width, height, imageType, bit_depth, 0, CV_8U, FrameMetadata::HOST, true));
+		}
+		createPipeline(inpFilePath, width, height,  imageType, bit_depth,  angle,  x,  y,  scale);
+	}
+	void createPipeline(const std::string& inpFilePath, int width, int height, ImageMetadata::ImageType imageType, int bit_depth, int angle, int x, int y, float scale)
+	{
+		fileReader = boost::shared_ptr<FileReaderModule>(new FileReaderModule(FileReaderModuleProps(inpFilePath)));	
+		auto rawImagePin = fileReader->addOutputPin(metadata);
+
+		auto stream = cudastream_sp(new ApraCudaStream);
+		copy1 = boost::shared_ptr<Module>(new CudaMemCopy(CudaMemCopyProps(cudaMemcpyHostToDevice, stream)));
+		fileReader->setNext(copy1);
+
+		AffineTransformProps affineProps(stream, angle, x, y, scale);
+		affineTransform = boost::shared_ptr<Module>(new AffineTransform(affineProps));
+		copy1->setNext(affineTransform);
+		copy2 = boost::shared_ptr<Module>(new CudaMemCopy(CudaMemCopyProps(cudaMemcpyDeviceToHost, stream)));
+		affineTransform->setNext(copy2);
+		m3 = boost::shared_ptr<ExternalSinkModule>(new ExternalSinkModule());
+		copy2->setNext(m3);
+
+		BOOST_TEST(fileReader->init());
+		BOOST_TEST(copy1->init());
+		BOOST_TEST(affineTransform->init());
+		BOOST_TEST(copy2->init());
+		BOOST_TEST(m3->init());
+	}
+	boost::shared_ptr<FileReaderModule>fileReader;
+	boost::shared_ptr<Module>copy1;
+	boost::shared_ptr<Module>affineTransform;
+	boost::shared_ptr<Module>copy2;
+	boost::shared_ptr<ExternalSinkModule>m3;
+
+	~AffineTestsStruct()
+	{
+		m3->term();
+		copy2->term();
+		affineTransform->term();
+		copy1->term();
+		fileReader->term();
+	}
+};
+BOOST_AUTO_TEST_CASE(MONO_rotation, *boost::unit_test::disabled())
```

Comment:
why is this test disabled ?

Reply:
i had done shift_scale_rotate all together in one test case so i disables those individual test cases